packages/components/src/disabled/test/index.tsx

```diff
@@ -0,0 +1,153 @@
+		expect( input ).toBeDisabled();
+		expect( div ).toHaveAttribute( 'contenteditable', 'false' );
+		expect( div ).not.toHaveAttribute( 'tabindex' );
+		expect( div ).not.toHaveAttribute( 'disabled' );
+	} );
+
+	it( 'should cleanly un-disable via reconciliation', () => {
+		const MaybeDisable = ( { isDisabled = true } ) =>
+			isDisabled ? (
+				<Disabled>
+					<Form />
+				</Disabled>
+			) : (
+				<Form />
+			);
+
+		const { container, rerender } = render( <MaybeDisable /> );
+		rerender( <MaybeDisable isDisabled={ false } /> );
+
+		const input = container.querySelector( 'form input' );
+		const div = container.querySelector( 'form div' );
+
+		expect( input ).not.toBeDisabled();
+		expect( div ).toHaveAttribute( 'contenteditable', 'true' );
+		expect( div ).toHaveAttribute( 'tabindex' );
+	} );
+
+	it( 'will disable or enable descendant fields based on the isDisabled prop value', () => {
+		const MaybeDisable = ( { isDisabled = true } ) => (
+			<Disabled isDisabled={ isDisabled }>
+				<Form />
+			</Disabled>
+		);
+
+		const { container, rerender } = render( <MaybeDisable /> );
+
+		const input = container.querySelector( 'form input' );
+		const div = container.querySelector( 'form div' );
+
+		expect( input ).toBeDisabled();
+		expect( div ).toHaveAttribute( 'contenteditable', 'false' );
+
+		rerender( <MaybeDisable isDisabled={ false } /> );
+
+		expect( input ).not.toBeDisabled();
+		expect( div ).toHaveAttribute( 'contenteditable', 'true' );
+	} );
+
+	// Ideally, we'd have two more test cases here:
+	//
+	//  - it( 'will disable all fields on component render change' )
+	//  - it( 'will disable all fields on sneaky DOM manipulation' )
+	//
+	// Alas, JSDOM does not support MutationObserver:
+	//
+	//  https://github.com/jsdom/jsdom/issues/639
+
+	describe( 'Consumer', () => {
+		function DisabledStatus() {
+			return (
+				<p>
+					<Disabled.Consumer>
+						{ ( isDisabled ) =>
+							isDisabled ? 'Disabled' : 'Not disabled'
+						}
+					</Disabled.Consumer>
+				</p>
+			);
+		}
+
+		test( "lets components know that they're disabled via context", () => {
+			const { container } = render(
+				<Disabled>
+					<DisabledStatus />
+				</Disabled>
+			);
+
+			const wrapperElement = container.querySelector( 'p' );
+			expect( wrapperElement ).toHaveTextContent( 'Disabled' );
+		} );
+
+		test( "lets components know that they're not disabled via context when isDisabled is false", () => {
+			const { container } = render(
+				<Disabled isDisabled={ false }>
+					<DisabledStatus />
+				</Disabled>
+			);
+			const wrapperElement = container.querySelector( 'p' );
+			expect( wrapperElement ).toHaveTextContent( 'Not disabled' );
+		} );
+
+		test( "lets components know that they're not disabled via context", () => {
+			const { container } = render( <DisabledStatus /> );
+			const wrapperElement = container.querySelector( 'p' );
+			expect( wrapperElement ).toHaveTextContent( 'Not disabled' );
+		} );
+	} );
```

Comment:
Let's not test implementation details by querying the DOM directly — we can use `screen.getByText` instead:

<details>

<summary>Click to expand</summary>

```diff
diff --git a/packages/components/src/disabled/test/index.tsx b/packages/components/src/disabled/test/index.tsx
index a4dcfb0f7d..b3c4d3728a 100644
--- a/packages/components/src/disabled/test/index.tsx
+++ b/packages/components/src/disabled/test/index.tsx
@@ -1,7 +1,7 @@
 /**
  * External dependencies
  */
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
 
 /**
  * Internal dependencies
@@ -124,30 +124,29 @@ describe( 'Disabled', () => {
 		}
 
 		test( "lets components know that they're disabled via context", () => {
-			const { container } = render(
+			render(
 				<Disabled>
 					<DisabledStatus />
 				</Disabled>
 			);
 
-			const wrapperElement = container.querySelector( 'p' );
-			expect( wrapperElement ).toHaveTextContent( 'Disabled' );
+			expect( screen.getByText( 'Disabled' ) ).toBeInTheDocument();
 		} );
 
 		test( "lets components know that they're not disabled via context when isDisabled is false", () => {
-			const { container } = render(
+			render(
 				<Disabled isDisabled={ false }>
 					<DisabledStatus />
 				</Disabled>
 			);
-			const wrapperElement = container.querySelector( 'p' );
-			expect( wrapperElement ).toHaveTextContent( 'Not disabled' );
+
+			expect( screen.getByText( 'Not disabled' ) ).toBeInTheDocument();
 		} );
 
-		test( "lets components know that they're not disabled via context", () => {
-			const { container } = render( <DisabledStatus /> );
-			const wrapperElement = container.querySelector( 'p' );
-			expect( wrapperElement ).toHaveTextContent( 'Not disabled' );
+		test( "lets components know that they're not disabled via context when the Disabled component is not rendered at all", () => {
+			render( <DisabledStatus /> );
+
+			expect( screen.getByText( 'Not disabled' ) ).toBeInTheDocument();
 		} );
 	} );
 } );

```
</details>